### PR TITLE
Add containerized retail workflow

### PR DESCRIPTION
**Pull request by aaannz**
# Description

Rework retail documentation to cover different configuration ways by including example configurations.

Include instructions how to configure retail using containerized proxy in the examples.

# Target branches

Which documentation version does this PR apply to?

- [X] Master (Default)
- [ ] Manager-4.3 https://github.com/uyuni-project/uyuni-docs/pull/2090
- [ ] Manager-4.2

# Links

Fixes https://github.com/SUSE/spacewalk/issues/19844

### CODE REVIEW

**Comment by aaannz** [comment]
This is ready for review, I need to once more validate if all described scenarios work as documented though.

**Comment by aaannz** [comment]
The rest of failed unit tests seems to be unrelated to this PR

**Comment by aaannz** [comment]
CC @keichwa

File: modules/retail/pages/retail-install-setup.adoc [hunk 1]
```diff
@@ -62,12 +68,46 @@ SLES{nbsp}11 does not support later versions of TLS.
...
+You can create system groups using the {productname} {webui}.
+Navigate to menu:Systems[System Groups] and click btn:[Create System Group].
+
+For more information about system groups, see xref:reference:systems/system-groups.adoc[].
+
+During terminal bootstrap terminal automatically joins:
+
+- branch system group based on received [systemitem]``branch_id``. This will make branch group formulas available to the terminal.
+- HWType group based on SMBios information received from terminal. This will make saltboot partitioning pillar available to the terminal.
+- [systemitem]``TERMINALS`` if this group exists
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:104]:
```suggestion
- [systemitem]``TERMINALS`` if this group exists.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 2]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
-Configure branch server services from the {productname} Server.
-The configuration is then applied to the selected branch server using Salt states.
-{productname} Formulas with Forms functionality is used to configure branch server services.
-However, there are multiple ways to configure these services:
+== Configure Services for Saltboot
 
-* {productname} {smr} provided command line tool [command]``retail_branch_init``
-* {productname} {smr} provided mass import command line tool [command]``retail_yaml``
-* {productname} web UI and configuring formulas manually (for advanced users)
+Saltboot is mechanism which is used to deploy POS images to the terminals. Saltboot consists of saltboot enabled initrd (build as part of POS images) and saltboot salt states.
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:126]:
```suggestion
Saltboot technology is used to deploy POS images to the terminals.
Saltboot consists of saltboot enabled initrd (build as part of POS images) and saltboot Salt states.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 3]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
+This section covers general information about generic Saltboot requirements.
+For configuration examples, see xref:retail:example-configurations[].
 
-The branch server can be configured automatically using the [command]``retail_branch_init`` command, as shown in this section.
-If you prefer to manually configure the branch server, you can do so using formulas.
-For more information about formulas, see xref:retail:retail-formulas-intro.adoc[].
+=== Enable PXE network boot in the terminal network
 
-.Procedure: Configuring Branch Server Formulas With a Helper Script
+Saltboot is usually used in network boot environment. For this to work [systemitem]``DHCP`` service for the network terminal is connected to must have [systemitem]``PXE`` or sometimes called [systemitem]``BOOTP`` support enabled.
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:133]:
```suggestion
Saltboot is usually used in network boot environment.
For this to work [systemitem]``DHCP`` service for the network terminal is connected to must have [systemitem]``PXE`` or sometimes called [systemitem]``BOOTP`` support enabled.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 4]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
-----
-salt <branch_server_minion_id> state.apply
-----
+Notice two important options, [systemitem]``next-server`` which is set to the branch server IP address and [systemitem]``filename`` set to the [systemitem]``pxelinux.0`` for BIOS based system and [systemitem]``shim.efi`` for UEFI systems with SecureBoot support.
 
-Similar results can be achieved by using mass import command line tool.
+=== Saltboot service discovery
 
-.Procedure: Configuring Branch Server Formulas With a Mass Import Tool
+Saltboot requires some information where is the salt master and where to download image from. Saltboot tries multiple discoveries to obtain this information, described below.
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:157]:
```suggestion
Saltboot requires some information where the Salt master is and from where to download the image.
Saltboot tries multiple discoveries to obtain this information, described below.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 5]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
-+
-For more information about mass import tool, see xref:retail:retail-mass-config.adoc[].
-. Import branch information from YAML file to {productname}
-+
-----
-retail_yaml --from-yaml branch.yaml
-----
-. Verify that your changes have been configured correctly by checking the {productname} {webui} branch server system formulas.
-. Apply highstate on the branch server.
+For successful terminal deployment, both service discoveries must be successful. Depending on your architecture, choose what strategy works for you best.
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:159]:
```suggestion
For successful terminal deployment, both service discoveries must be successful.
Depending on your architecture, choose what strategy works best for you.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 6]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
 
-[WARNING]
-====
-Both [command]``retail_branch_init`` and [command]``retail_yaml`` commands override existing configuration settings of the specified branch server.
-====
+==== Salt master discovery
 
-After the initial configuration done by command line tools, branch server configuration can be further adjusted in {productname} {webui} through branch server formulas.
+During saltboot initrd start, integrated salt minion needs to find branch server proxy to connect to. This discovery is trying following steps:
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:163]:
```suggestion
During saltboot initrd start, integrated Salt client needs to find branch server proxy to connect to.
This discovery is trying following steps:
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 7]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
 
-=== Required System Groups
+- [systemitem]``MASTER`` kernel command line option is set, then this is used as salt master
+- resolve [systemitem]``salt`` CNAME, if successful then resolved value is used as salt master
+- use [systemitem]``salt`` as a salt master
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:167]:
```suggestion
- [systemitem]``MASTER`` kernel command line option is set, then this is used as Salt master
- resolve [systemitem]``salt`` CNAME, if successful then resolved value is used as Salt master
- use [systemitem]``salt`` as a Salt master
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 8]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
 
-{productname} {smr} requires system groups for terminals and servers.
-Manually create these system groups during installation:
+Once salt master is determined, salt minion configuration is generated and started.
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:169]:
```suggestion
When Salt master is determined, Salt client configuration is generated and started.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 9]
```diff
@@ -81,118 +121,77 @@ In case you plan to use the branch server with Ansible software, be aware that A
...
-[NOTE]
-====
-{productname} {smr} command line tools create required system groups and branch group automatically.
-====
+Value obtained by download server discovery is then used together with POS image pillar to fetch correct image from correct location.
+
+=== Terminal partitioning and image selection
+
+Last piece for Saltboot is to provide partitioning for terminal. This is done individually for each hardware type of terminals.
+For more information about hardware types, see xref:retail:retail-deploy-terminals[]
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:191]:
```suggestion
Last piece for Saltboot is to provide partitioning for terminal.
This is done individually for each hardware type of terminals.
For more information about hardware types, see xref:retail:retail-deploy-terminals[].
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 10]
```diff
@@ -0,0 +1,134 @@
+[[shared-central-dns]]
+= Set Up the {productname} {smr} Environment with shared network between terminals, branch servers and {productname}
+
+To set up the {productname} {smr} environment, you will need to have already installed and configured {productname} Server, have one or more {productname} {smr} branch server proxies installed and one of more OS images built.
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:4]:
```suggestion
To set up the {productname} {smr} environment, you will need to have already:

 * {productname} Server installed and configured
 * one or more {productname} {smr} branch server proxies installed
 * one of more OS images built.
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+In this example we provide [systemitem]``salt`` on kernel command line option and [systemitem]``ftp`` in pillar for saltboot service discovery.
+
+We assume to have one regular branch server with salt minion id [systemitem]``branch1.mystore.com``, which is equal to the branch server fully qualified domain name, and branch id [systemitem]``B0001``.
+
+As a terminal we assume to have one terminal with hardware manufacturer [systemitem]``TerminalOEM`` and model [systemitem]``T1000``.
+
+For POS image we assume to have one with name [systemitem]``POS_Image_JeOS7``.
+
+There is already existing external DHCP and DNS service. DHCP service is correctly configured to PXE boot from branch server.
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:18]:
```suggestion
There is already existing external DHCP and DNS service.
DHCP service is correctly configured to PXE boot from branch server.
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+== Create required system groups
+
+Following guide xref:reference:systems/system-groups.adoc[] we create following system groups:
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:22]:
```suggestion
Follow xref:reference:systems/system-groups.adoc[] to create the following system groups:
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+- Tftpd
+
+.Procedure: Assign formulas to the branch server
+
+. Select the [guimenu]``Formulas`` tab for the system [systemitem]``branch1.mystore.com``
+. You should see list of formulas in [guimenu]``Configuration`` sub-tab
+. In displayed list select [systemitem]``Branch Network``
+. In displayed list select [systemitem]``Pxe``
+. In displayed list select [systemitem]``Tftpd``
+. Save assigned formulas by clicking on [guimenu]``Save`` button in top right corner
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:48]:
```suggestion
. Save assigned formulas by clicking [guimenu]``Save`` in top right corner
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+.Procedure: Configure Tftpd formula
+
+. Select Tftpd formula from branch server formula list
+. Set [systemitem]``Internal Network Address`` to [systemitem]``0.0.0.0``, this way tftp will be listening on any IPv4
+. Set [systemitem]``TFTP base directory`` to [systemitem]``/srv/saltboot``
+. Save the formula
+
+[NOTE]
+====
+Next two formulas are group formulas. Make sure you are editing those formulas in their respective groups.
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:74]:
```suggestion
Next two formulas are group formulas
Make sure you are editing those formulas in their respective groups.
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+Assign [systemitem]``Saltboot Group`` formula to the [systemitem]``B0001`` group and [systemitem]``Saltboot`` formula to the [systemitem]``HWType:TerminalOEM-T1000`` group.
+
+.Procedure: Assign Saltboot Group formula to Branch group
+
+. Navigate to menu:Systems[System Groups]
+. Select group [systemitem]``B0001`, which is our branch group
+. Select the [guimenu]``Formulas`` tab once in group details
+. You should see list of formulas in [guimenu]``Configuration`` sub-tab
+. In displayed list select [systemitem]``Saltboot Group``
+. Save assigned formulas by clicking on [guimenu]``Save`` button in the top right corner
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:88]:
```suggestion
. Save assigned formulas by clicking [guimenu]``Save`` in the top right corner
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+. Save the formula
+
+.Procedure: Assign Saltboot formula to hardware type group
+
+. Navigate to menu:Systems[System Groups]
+. Select group [systemitem]``HWType:TerminalOEM-T1000`, which is our hardware type group
+. Select the [guimenu]``Formulas`` tab once in group details
+. You should see list of formulas in [guimenu]``Configuration`` sub-tab
+. In displayed list select [systemitem]``Saltboot``
+. Save assigned formulas by clicking on [guimenu]``Save`` button in top right corner
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:104]:
```suggestion
. Save assigned formulas by clicking [guimenu]``Save`` in the top right corner
```

File: modules/retail/pages/shared-central-dns.adoc [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+. Set [systemitem]``Filesystem Format`` to [systemitem]``vfat``
+. Set [systemitem]``Partition Flags`` to [systemitem]``boot``
+. Click [systemitem]``+`` to add a partition
+. Set [systemitem]``Partition Symbolic ID`` to [systemitem]``p2``
+. Set [systemitem]``Device Mount Point`` to [systemitem]``/``
+. Set [systemitem]``OS Image to Deploy`` to [systemitem]``POS_Image_JeOS7``
+. Save the formula
+
+
+After all procedures are done, apply highstate on the branch server
```
Comment (keichwa) [inline modules/retail/pages/shared-central-dns.adoc:126]:
```suggestion
After all procedures are done, apply highstate on the branch server.
```

**Review by keichwa — CHANGES_REQUESTED** [review]
This was huge :)
Here mostly just cosmetic change proposals.  Later, I'll rebuild the book and check the visual results.

File: modules/retail/pages/containerized-saltboot.adoc [hunk 18]
```diff
@@ -63,65 +70,45 @@ See xref:retail-terminal-names.adoc[] for explanation of possible configurations
...
 Containerized workflow relies on updated image building in {productname} Server 4.3 where PXE images are no longer collected as bundle, but kernel, initrd and filesystem image are collected individually.
 
 Containerized workflow uses new TFTP container which instead of providing files present on the proxy, routes specific TFTP requests as HTTP requests through proxy local squid proxy to {productname} Server.
 
-Leveraging these two improvements and the fact that containerized proxy is not a salt client, it is no longer possible to call [systemitem]``image-sync`` state.
-After image is build images are immediately available to the saltboot clients. This may have implications on how images are deployed to production.
+Containerized proxy is not a Salt client, it is not possible to call [systemitem]``image-sync`` state.
 
-Following documents differences between containerized and regular workflow. Both are assuming proxy (containerized or in form of {smr} Branch Server) are available.
+Thanks to these improvements and not present Satl client, after image is build it is immediately available to the Saltboot clients.
```
Comment (keichwa) [inline modules/retail/pages/containerized-saltboot.adoc:83]: It isn't completely clear to me what this statement (the first half of the sentence) actually means.  Maybe, a suggestion I made caused the trouble.

File: modules/retail/pages/containerized-saltboot.adoc [hunk 19]
```diff
@@ -63,65 +70,45 @@ See xref:retail-terminal-names.adoc[] for explanation of possible configurations
...
+This may have implications on how images are deployed to production.
+
+The following sections difference between containerized and regular workflow.
```
Comment (keichwa) [inline modules/retail/pages/containerized-saltboot.adoc:86]:
```suggestion
The following sections differentiate between containerized and regular workflow.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 20]
```diff
@@ -59,19 +63,21 @@ POS Terminals based on {sles}{nbsp}11 SP{nbsp}3 can be deployed in much the same
...
 ====
 
-== Required System Groups
+== Branch identification and architecture topology
 
-Before you configure the branch server, ensure you have decided on networking topology and you choose [systemitem]``branch id`` for every branch.
-For the information about the possible network topologies, see xref:retail:retail-network-arch.adoc[]. As a [systemitem]``branch id`` select any alphanumerical string.
+Before you configure the branch server, ensure you have decided on networking topology and you choose [systemitem]``branch id``.
+
+For the information about the possible network topologies, see xref:retail:retail-network-arch.adoc[].
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:76]:
```suggestion
For information about the possible network topologies, see xref:retail:retail-network-arch.adoc[].
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 21]
```diff
@@ -123,14 +129,16 @@ For more information about installing Ansible, see xref:administration:ansible-i
 
 == Configure Services for Saltboot
 
-Saltboot is mechanism which is used to deploy POS images to the terminals. Saltboot consists of saltboot enabled initrd (build as part of POS images) and saltboot salt states.
+Saltboot technology is used to deploy POS images to the terminals.
+Saltboot consists of saltboot enabled initrd (build as part of POS images) and saltboot salt states.
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:133]:
```suggestion
Saltboot consists of saltboot enabled initrd (build as part of POS images) and saltboot Salt states.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 22]
```diff
@@ -187,11 +198,12 @@ Value obtained by download server discovery is then used together with POS image
 
 === Terminal partitioning and image selection
 
-Last piece for Saltboot is to provide partitioning for terminal. This is done individually for each hardware type of terminals.
-For more information about hardware types, see xref:retail:retail-deploy-terminals[]
+Last piece for Saltboot is to provide partitioning for terminal.
+This is done individually for each hardware type of terminals.
+For more information about hardware types, see xref:retail:retail-deploy-terminals.adoc[]
```
Comment (keichwa) [inline modules/retail/pages/retail-install-setup.adoc:203]:
```suggestion
For more information about hardware types, see xref:retail:retail-deploy-terminals.adoc[].
```

**Review by keichwa — APPROVED** [review]
Some more cosmetic things.  Thanks!

File: modules/retail/pages/containerized-saltboot.adoc (hunk 18 above)
Comment (aaannz) [reply modules/retail/pages/containerized-saltboot.adoc]: I removed first sentence and focus on what is important: build image is immediately available to the clients without `image-sync` application.

File: modules/retail/pages/containerized-saltboot.adoc [hunk 23]
```diff
@@ -0,0 +1,158 @@
+[[retail-install-setup-containerized]]
+= Set Up the {productname} {smr} Environment using containerized proxy
+
+To set up the {productname} {smr} environment, you will need to have already installed and configured:
+
+* {productname} {smr} Server 4.3 or newer
+* one or more {productname} {smr} containerized proxies
+* one or more {productname} build host
+
+This section covers how to configure your  environment using containerized {productname} Proxy for [systemitem]``Saltboot`` deployment.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:10]:
```suggestion
The only remaining step to configure is a SaltBoot Group.

This section covers how to configure your environment using containerized {productname} Proxy for [systemitem]``Saltboot Group`` deployment.
```

File: modules/retail/pages/containerized-saltboot.adoc [hunk 24]
```diff
@@ -0,0 +1,158 @@
...
+See xref:specialized-guides:salt/salt-formula-dhcpd.adoc[] how to use DHCP formula to configure DHCP server.
+
+Without using DHCP formula, make sure your DHCP server has correct PXE booting setting pointing to containerized proxy.
+See below for example.
+====
+
+== Saltboot group
+
+Containerized {productname} {smr} is configured in [systemitem]``Saltboot Group``.
+Saltboot Groups are regular server groups with [systemitem]``Saltboot Group`` formula enabled.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:35]:
```suggestion
Saltboot Groups are regular system groups with [systemitem]``Saltboot Group`` formula enabled.

The first step is to create a new System Group, from Systems Menu.
This system group must have enabled the Formula "Saltboot Group" from the Formulas tab.
```

File: modules/retail/pages/containerized-saltboot.adoc [hunk 25]
```diff
@@ -0,0 +1,158 @@
...
+
+=== Naming scheme for new registrations
+
+Last three options are related to how will be newly registered machine visible in {productname} Server.
+
+See xref:retail-terminal-names.adoc[] for explanation of possible configurations.
+
+== Comparing containerized and non-containerized workflows
+
+External DHCP service must be used with containerized Saltboot.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:71]: Wow. That could be a big constraint.
And we don't even provide any guide to help the user in this.. let's call it "regression" comparing to what we were offering with the regular retail branch server.

Comment (srbarrios) [reply modules/retail/pages/containerized-saltboot.adoc:71]: OK, @aaannz clarified that most of our customers are not using the DHCP coming from the Branch server as explained in our Documentation. But instead, they use their own DHCP.
If that's the case for the majority of our customers, then I guess this is not such a big deal.

File: modules/retail/pages/containerized-saltboot.adoc [hunk 26]
```diff
@@ -0,0 +1,158 @@
...
+Older images will not work.
+====
+
+[IMPORTANT]
+====
+Containerized workflow not longer implicitly configure DHCP for PXE booting.
+See xref:specialized-guides:salt/salt-formula-dhcpd.adoc[] how to use DHCP formula to configure DHCP server.
+
+Without using DHCP formula, make sure your DHCP server has correct PXE booting setting pointing to containerized proxy.
+See below for example.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:29]: Missing example?

File: modules/retail/pages/containerized-saltboot.adoc [hunk 27]
```diff
@@ -0,0 +1,158 @@
...
+
+== Comparing containerized and non-containerized workflows
+
+External DHCP service must be used with containerized Saltboot.
+
+Containerized workflow relies on updated image building in {productname} Server 4.3 where PXE images are no longer collected as bundle, but kernel, initrd and filesystem image are collected individually.
+
+Containerized workflow uses new TFTP container which instead of providing files present on the proxy, routes specific TFTP requests as HTTP requests through local squid proxy to {productname} Server.
+
+Containerized proxy is not a Salt client, it is not possible to call [systemitem]``image-sync`` state.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:77]: I would extend this, by explaining how we should synchronize new images now.

File: modules/retail/pages/containerized-saltboot.adoc [hunk 28]
```diff
@@ -0,0 +1,158 @@
...
+
+See xref:retail-terminal-names.adoc[] for explanation of possible configurations.
+
+== Comparing containerized and non-containerized workflows
+
+External DHCP service must be used with containerized Saltboot.
+
+Containerized workflow relies on updated image building in {productname} Server 4.3 where PXE images are no longer collected as bundle, but kernel, initrd and filesystem image are collected individually.
+
+Containerized workflow uses new TFTP container which instead of providing files present on the proxy, routes specific TFTP requests as HTTP requests through local squid proxy to {productname} Server.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:75]: Do we list which specific requests somewhere?

File: modules/retail/pages/containerized-saltboot.adoc [hunk 29]
```diff
@@ -0,0 +1,158 @@
...
+====
+
+The following sections differentiate between containerized and regular workflow.
+Both are assuming proxy (containerized or in form of {productname} {smr} Branch Server) are available.
+
+Containerized workflow:
+
+. Build POS image
+. Configure DHCP server for PXE booting for given network
+. Create [systemitem]``Saltboot group`` and configure it for existing containerized proxy
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:92]: This section miss the external DHCP, and the need of a system for saltboot formula.

**Comment by 0rnela** [comment]
@aaannz - can you please take a look and implement @srbarrios' suggested changes, then resubmit for approval?

**Comment by aaannz** [comment]
> @aaannz - can you please take a look and implement @srbarrios' suggested changes, then resubmit for approval?

Next week

**Comment by srbarrios** [comment]
If it helps, these are the steps I'm doing to accomplish that workflow (in-progress):


**Steps**
- Configure and start the Containerized Proxy (We re-use the Proxy VM for that purpose, so we must stop all proxy services from regular proxy, including dhcpd and named):
  ```
  spacewalk-proxy stop
  systemctl stop dhcpd
  systemctl stop named
  systemctl stop firewalld
  systemctl start uyuni-proxy-pod
  ```
- Create a new System Group, from Systems Menu. This system group must have enabled the Formula `Saltboot Group` from the `Formulas` tab
  
- Create a all the necessary System Groups, from Systems Menu. These system groups must have enabled the Formula `Saltboot` from the `Formulas` tab
  
  

- Create a new VM that will contain a external DHCP server (as the new Containerized Proxy doesn't support DHCP and DNS services configuration management):
  ```
  zypper in dhcp-relay dhcp-server dhcp-tools
  ```
- Enable the private network in that VM (as the Pod Proxy is in a VM using 192.168.43.254, we are force to use 192.168.43.253):
  ```
  mgr197:/etc/sysconfig/network # cat ifcfg-eth1
  BOOTPROTO='static'
  STARTMODE='hotplug'
  ZONE='public'
  IPADDR='192.168.43.253'
  NETMASK='255.255.255.0'
  mgr197:/etc/sysconfig/network # sudo wicked ifup eth1
  eth1            up
  mgr197:/etc/sysconfig/network # wicked ifstatus eth1
  eth1            up
        link:     #3, state up, mtu 1500
        type:     ethernet, hwaddr 52:54:00:4d:24:4a
        config:   compat:suse:/etc/sysconfig/network/ifcfg-eth1
        leases:   ipv4 static granted
        addr:     ipv4 192.168.43.253/24 [static]
  ```
- Configure the DHCP server on that VM attached to the private network (note we only configure IPv4):
  ```
  mgr197:/etc # cat /etc/sysconfig/dhcpd
  DHCPD_INTERFACE="eth1"
  ```
  ```
  mgr197:~ # cat /etc/dhcpd.conf 
  # dhcpd.conf
  #
  # SaltStack-generated configuration file for ISC dhcpd
  #
  # option definitions common to all supported networks...
  option domain-name "example.org";
  option domain-name-servers 192.168.43.253;
  option arch code 93 = unsigned integer 16; # RFC4578
  
  default-lease-time 20000;
  max-lease-time 20001;
  
  #use-host-decl-names off;
  
  # Use this to enble / disable dynamic dns updates globally.
  #ddns-update-style none;
  #update-static-leases off;
  
  # If this DHCP server is the official DHCP server for the local
  # network, the authoritative directive should be uncommented.
  authoritative;
  
  # Use this to send dhcp log messages to a different log file (you also
  # have to hack syslog.conf to complete the redirection).
  subnet 192.168.43.0 netmask 255.255.255.0 {
    range 192.168.43.128 192.168.43.252;
    option broadcast-address 192.168.43.255;
    if option arch = 00:0b {
      filename "boot/grub-arm64.efi";
      next-server 192.168.43.253;
    }
    elsif option arch = 00:07 {
       filename "boot/shim.efi";
       next-server 192.168.43.253;
    }
    else {
      filename "boot/pxelinux.0";
      next-server 192.168.43.253;
    }
  }
  
  
  host sle12sp5terminal {
    hardware ethernet 52:54:00:3A:8D:2C;
    fixed-address 192.168.43.5;
  }
  
  host sle15sp4terminal {
    hardware ethernet 52:54:00:7A:3A:80;
    fixed-address 192.168.43.6;
  }
  ```
- Start the DHCPD service and check it:
  ```
  mgr197:~ # systemctl start dhcpd
  mgr197:~ # systemctl status dhcpd
  ● dhcpd.service - ISC DHCPv4 Server
     Loaded: loaded (/usr/lib/systemd/system/dhcpd.service; disabled; vendor preset: disabled)
     Active: active (running) since Thu 2023-02-09 10:12:30 CET; 4s ago
    Process: 9737 ExecStart=/usr/lib/dhcp/dhcpd -4 start (code=exited, status=0/SUCCESS)
   Main PID: 9832 (dhcpd)
      Tasks: 1
     CGroup: /system.slice/dhcpd.service
             └─9832 /usr/sbin/dhcpd -4 -cf /etc/dhcpd.conf -pf /var/run/dhcpd.pid -chroot /var/lib/dhcp -lf /db/dhcpd.leases -user dhcpd -group nogroup eth1
  
  feb 09 10:12:30 mgr197 dhcpd[9831]: PID file: /var/run/dhcpd.pid
  feb 09 10:12:30 mgr197 dhcpd[9831]: Wrote 0 deleted host decls to leases file.
  feb 09 10:12:30 mgr197 dhcpd[9831]: Wrote 0 new dynamic host decls to leases file.
  feb 09 10:12:30 mgr197 dhcpd[9831]: Wrote 0 leases to leases file.
  feb 09 10:12:30 mgr197 dhcpd[9831]: Listening on LPF/eth1/52:54:00:4d:24:4a/192.168.43.0/24
  feb 09 10:12:30 mgr197 dhcpd[9831]: Sending on   LPF/eth1/52:54:00:4d:24:4a/192.168.43.0/24
  feb 09 10:12:30 mgr197 dhcpd[9831]: Sending on   Socket/fallback/fallback-net
  feb 09 10:12:30 mgr197 dhcpd[9832]: Server starting service.
  feb 09 10:12:30 mgr197 dhcpd[9737]: Starting ISC DHCPv4 Server [chroot]..done
  feb 09 10:12:30 mgr197 systemd[1]: Started ISC DHCPv4 Server.
  ```

File: modules/retail/pages/containerized-saltboot.adoc [hunk 30]
```diff
@@ -0,0 +1,158 @@
...
+[IMPORTANT]
+====
+Containerized workflow requires POS images build using {productname} Server 4.3 or newer.
+Older images will not work.
+====
+
+[IMPORTANT]
+====
+Containerized workflow not longer implicitly configure DHCP for PXE booting.
+See xref:specialized-guides:salt/salt-formula-dhcpd.adoc[] how to use DHCP formula to configure DHCP server.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:26]: We might mention the possible need of DNS formula too.
At least, I required it in my manual setup.

File: modules/retail/pages/containerized-saltboot.adoc [hunk 31]
```diff
@@ -0,0 +1,158 @@
...
+
+Configure [systemitem]``Default boot image for new registrations`` to specify what boot image should be booted by not yet registered POS terminal.
+This is useful when stable boot image is wanted for initial deployments.
+Without this setting, newest built boot image is used as default boot image.
+
+If [systemitem]``Default boot image for new registrations`` is set, option to set its version appears under name [systemitem]``Default boot image version`` where specific image version can be set.
+
+=== Kernel option for the saltboot group
+
+Option [systemitem]``Kernel parameters for the group`` can be used to pass extra kernel options to all POS terminals registered withing this saltboot group.
```
Comment (srbarrios) [inline modules/retail/pages/containerized-saltboot.adoc:61]: We should add a comment here to troubleshooting the issue I add, solved by included `id=dhcp`

File: modules/retail/pages/containerized-saltboot.adoc (hunk 23 above)
Comment (aaannz) [reply modules/retail/pages/containerized-saltboot.adoc:10]: Added

File: modules/retail/pages/containerized-saltboot.adoc (hunk 26 above)
Comment (aaannz) [reply modules/retail/pages/containerized-saltboot.adoc:29]: Link added

File: modules/retail/pages/containerized-saltboot.adoc (hunk 27 above)
Comment (aaannz) [reply modules/retail/pages/containerized-saltboot.adoc:101]: Added a note there is no image sync in containerized proxy.

**Comment by aaannz** [comment]
Updated with feedback from @srbarrios

File: .changelog [hunk 32]
```diff
@@ -4,6 +4,7 @@
 # - Fixed error in Bat section of Upgrade Guide (bsc#1234567)
 # For guidelines: https://en.opensuse.org/openSUSE:Creating_a_changes_file_(RPM)
 
+- Rework Retail documentaion to have generic config + examples
```
Comment (keichwa) [inline .changelog:7]:
```suggestion
- Rework Retail documentation to have generic configuration and examples
```

**Review by keichwa — APPROVED** [review]

**Pull request merged**